Gems/LandscapeCanvas/Code/Source/Editor/MainWindow.cpp

```diff
@@ -226,6 +227,22 @@ namespace LandscapeCanvasEditor
 
         AZ_Assert(categoryIt->second.size(), "No components found that satisfy the missing required service(s).");
 
+        const AZStd::string categoryName(categoryIt->first.toUtf8());
+
+        // Check whether the selected category has a preferred component and return that if it does.
+        for (const auto& preferredComponentPair : preferredComponentByCategory)
+        {
+            if (categoryName == preferredComponentPair.first)
+            {
+                const auto& componentPair = categoryIt->second.find(preferredComponentPair.second);
+
+                if (componentPair != categoryIt->second.end())
+                {
+                    return componentPair->second->m_typeId;
+                }
+            }
+        }
+
```

Comment:
This might _work_ for the one example we have, but I don't think it's correct.  I believe the way this method works is that it has an input set of categories and component names that have already been filtered down to be all the potential matches for the component service we're trying to match.  So, this method is trying to pick the best component out of that set.
What I think we _want_ is to say "If we have a specific preferred component that exists in the set, return that.  If not, if we have a specific preferred category, pick the first component from that category.  If not, just return the first component that matches."
What I think we're saying with the above is "If we have a specific preferred category, look in that category for any of our preferred components.  If we don't, look in the first category for any of our preferred components.  Otherwise, return the first component in that category."
Basically, I think the preferredComponentByCategory loop should appear before the preferredCategory loop.  Also, since it's checking everything in the preferredComponentByCategory, that could just be a static const array instead of an unordered_map.  We're never performing lookups in it, so an array would be a better choice than a map.

Reply:
I can think of arguments for both ways round, but switching the order probably makes most sense. I've left it as a unordered_map, but put a lookup in instead, which is what I intended in the first place.